src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -251,13 +251,29 @@ bool CompositionalCouplingScheme::isActionRequired(
   return isRequired;
 }
 
+bool CompositionalCouplingScheme::isActionFulfilled(
+    const std::string &actionName) const
+{
+  PRECICE_TRACE(actionName);
+  bool isFulfilled = false;
+  for (const Scheme &scheme : _couplingSchemes) {
+    if (not scheme.onHold) {
+      isFulfilled |= scheme.scheme->isActionFulfilled(actionName);
```

Comment:
```suggestion
  bool isFulfilled = true;
  for (const Scheme &scheme : _couplingSchemes) {
    if (not scheme.onHold) {
      isFulfilled &= scheme.scheme->isActionFulfilled(actionName);
```
?
It's a bit worrisome that we don't catch this with a test

Reply:
This is actually correct.

Not all schemes require the action, so not all will have the action fulfilled.